src/components/admin/AdminLoungeMute.vue

```diff
@@ -0,0 +1,215 @@
+                  <v-col>
+                    <v-btn
+                      v-if="showConfirmation && isMuteEndDateSet"
+                      color="primary"
+                      class="w3-race-bg--text"
+                      @click="save"
+                    >
+                      {{ $t(`views_admin.ok`) }}
+                    </v-btn>
+                  </v-col>
+                  <v-col class="text-right">
+                    <v-btn
+                      color="primary"
+                      class="w3-race-bg--text"
+                      @click="close"
+                    >
+                      {{ $t(`views_admin.cancel`) }}
+                    </v-btn>
+                  </v-col>
+                </v-row>
+              </v-container>
+            </v-card-text>
+          </v-card>
+        </v-dialog>
+      </v-toolbar>
+    </template>
+
+    <template #[`item.actions`]="{ item }">
+      <v-icon small @click="deleteItem(item)">mdi-delete</v-icon>
+    </template>
+  </v-data-table>
+</template>
+
+<script lang="ts">
+import Vue from "vue";
+import { Component } from "vue-property-decorator";
+import { LoungeMute, LoungeMuteResponse } from "@/store/admin/loungeMute/types";
+import PlayerSearch from "@/components/common/PlayerSearch.vue";
+import { useOauthStore } from "@/store/oauth/store";
+import { setDate, setMonth, setYear } from "date-fns";
+
+@Component({ components: { PlayerSearch } })
+export default class AdminLoungeMute extends Vue {
+  public dateMenu = false;
+  public dialog = false;
+  public showConfirmation = false;
+  public battleTag = "";
+  public endDate = "";
+  private oauthStore = useOauthStore();
+
+  public headers = [
+    { text: "BattleTag", align: "start", sortable: true, value: "battleTag" },
+    { text: "Mute End Date", sortable: true, value: "endDate" },
+    { text: "Mute Insert Date", sortable: true, value: "insertDate" },
+    { text: "Author", sortable: true, value: "author" },
+    { text: "Actions", sortable: false, value: "actions" },
+  ];
+
+  get loungeMutes(): LoungeMuteResponse[] {
+    return this.$store.direct.state.loungeMute.loungeMutedPlayers;
+  }
+
+  async deleteItem(item: LoungeMute): Promise<void> {
+    confirm("Are you sure you want to delete this item?") &&
+    await this.$store.direct.dispatch.loungeMute.deleteLoungeMute(item.battleTag);
+    this.loadMutes();
+  }
+
+  get isMuteEndDateSet(): boolean {
+    return this.endDate != "";
+  }
+
+  public close(): void {
+    this.dialog = false;
+    this.showConfirmation = false;
+  }
+
+  public async save(): Promise<void> {
+    this.close();
+
+    const mute = {
+      battleTag: this.battleTag,
+      author: this.author,
+      endDate: this.muteEndDateTime,
+    } as LoungeMute;
+
+    await this.$store.direct.dispatch.loungeMute.addLoungeMute(mute);
+    this.loadMutes();
+  }
+
+  get muteEndDateTime(): string {
+    if (!this.endDate) return "";
+    const endDateArr = this.endDate.split("-");
```

Comment:
This doesn't look good. Why do we do it?

Reply:
To format a date formatted like yyyy-MM-dd to yyyy-mm-ddThh:mm:ssZ. So in other words a UTC time. It's a bit ugly, but it does the job. Reasonably endDate should never be anything but a string formatted like yyyy-MM-dd. But I did a safeguard anyway so if endDate isn't set, we send an empty string to backend, where the request is answered with a proper error code.